dotnet-interactive.Tests/KernelHttpApiTests.cs

```diff
@@ -66,23 +81,69 @@ public async Task variable_is_returned_with_application_json_content_type(Langua
         [InlineData(Language.FSharp)]
         public async Task When_variable_does_not_exist_then_it_returns_404(Language language)
         {
-            var response = await _server.HttpClient.GetAsync($"/variables/{language.LanguageName()}/does_not_exist");
+            var response = await GetServer(language).HttpClient.GetAsync($"/variables/{language.LanguageName()}/does_not_exist");
 
             response.StatusCode.Should().Be(HttpStatusCode.NotFound);
         }
 
         [Fact]
         public async Task When_subkernel_does_not_exist_then_it_returns_404()
         {
-            var response = await _server.HttpClient.GetAsync("/variables/does_not_exist/does_not_exist");
+            var response = await GetServer().HttpClient.GetAsync("/variables/does_not_exist/does_not_exist");
 
             response.StatusCode.Should().Be(HttpStatusCode.NotFound);
         }
 
+        [Theory]
+        [InlineData(Language.CSharp)]
+        public async Task lsp_textDocument_hover_returns_expected_placeholder(Language language)
+        {
+            var request = JObject.Parse(@"
+{
+    ""textDocument"": {
+        ""uri"": ""path/to/document""
+    },
+    ""position"": {
+        ""line"": 1,
+        ""character"": 2
+    }
+}
+");
+            var response = await GetServer(language).HttpClient.PostJsonAsync("/lsp/textDocument/hover", request);
+            response.EnsureSuccessStatusCode();
+            var responseJson = await response.Content.ReadAsStringAsync();
+            var json = JObject.Parse(responseJson);
+            var markdownContent = json["contents"]["value"].Value<string>();
+            markdownContent.Should().Be("textDocument/hover at position (1, 2) with `markdown`");
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        public async Task unimplemented_lsp_methods_return_empty_string(Language language)
+        {
+            // language kernels that implement LSP handling, but not for the specified method
+            var response = await GetServer(language).HttpClient.PostJsonAsync($"/lsp/not/a/method", new JObject());
+            response.EnsureSuccessStatusCode();
+            var responseJson = await response.Content.ReadAsStringAsync();
+            responseJson.Should().BeEmpty();
+        }
+
+        [Theory]
+        [InlineData(Language.FSharp)]
+        [InlineData(Language.PowerShell)]
+        public async Task unimplemented_lsp_handlers_return_empty_string(Language language)
+        {
+            // language kernels that don't implement any LSP handling
+            var response = await GetServer(language).HttpClient.PostJsonAsync($"/lsp/textDocument/hover", new JObject());
+            response.EnsureSuccessStatusCode();
```

Comment:
This should not be called in a test because the error won't contain useful information, e.g. why it failed and what was actually returned.

We have a `ShouldSucceed()` extension method in dotnet/try that's much more useful here.

Reply:
I think `EnsureSuccessStatusCode()` is sufficient here.  In dotnet/try I found [`ShouldSucceedWithOutput()`](https://github.com/dotnet/try/blob/ffec59d135585039792217ed14da25d7f7858d08/WorkspaceServer.Tests/RunResultExtensions.cs#L24) and all that does is verify that a 200 code was returned.  I temporarily made a test fail to see the output of `EnsureSuccessStatusCode()` and it was

```
System.Net.Http.HttpRequestException : Response status code does not indicate success: 404 (Not Found).
```

Which I think is as expressive as we can hope for here.

Reply:
In this case, at this point in time, maybe, but it won't be ideal as the code grows. As an API gets more interesting, the return code is rarely very informative, e.g. we'll want to know why a 400 or 500 was returned. The test should pinpoint the problem, not just indicate that there was one.

Reply:
The only thing I can think of is to also output the response body if there was one.  If that's what we'll want I can add that helper.

Reply:
Just added commit `report more helpful errors` that logs the return code as well as the response content (if any).